packages/dd-trace/src/debugger/devtools_client/source-maps.js

```diff
@@ -33,13 +34,25 @@ const self = module.exports = {
   }
 }
 
+// TODO: Remove if-statement around `setTimeout` below once it's safe to do so.
+//
+// This is a workaround for, what seems like a bug in Node.js core, that seems to trigger when, among other things, a
+// lot of timers are being created very rapidly. This makes the call to `setTimeout` throw an error from within
+// `AsyncLocalStorage._propagate` with the following error message:
+//
+//     TypeError: Cannot read properties of undefined (reading 'Symbol(kResourceStore)')
+//
+// Source: https://github.com/nodejs/node/blob/v18.20.6/lib/async_hooks.js#L312
 function cacheIt (key, value) {
-  clearTimeout(cacheTimer)
-  cacheTimer = setTimeout(function () {
-    // Optimize for app boot, where a lot of reads might happen
-    // Clear cache a few seconds after it was last used
-    cache.clear()
-  }, 10_000).unref()
+  if (Date.now() > cacheTimerLastSet + 1_000) {
+    clearTimeout(cacheTimer)
+    cacheTimer = setTimeout(function () {
+      // Optimize for app boot, where a lot of reads might happen
+      // Clear cache a few seconds after it was last used
+      cache.clear()
+    }, 10_000).unref()
+    cacheTimerLastSet = Date.now()
```

Comment:
no point in computing date.now() twice no ?

Reply:
Fixed 👍